lib/worldize.rb

```diff
@@ -40,39 +40,58 @@ def country_names
       @countries.map{|c| c.properties.name}
     end
 
-    # NB: syntax draw(countries = {}, **options) causes segfault in Ruby 2.2.0
-    def draw(countries_and_options = {})
-      options = DEFAULT_OPTIONS.merge(countries_and_options)
-      width = options.fetch(:width)
-      
-      img = Image.new(width, width){
-        self.background_color = options.fetch(:ocean)
-      }
-
-      gc = Magick::Draw.new.
-        stroke(options.fetch(:border)).stroke_width(1).
-        fill(options.fetch(:land))
-
-      @countries.each do |country|
-        bg = countries_and_options[country.properties.name] ||
-              countries_and_options[country.properties.iso_a3] ||
-              options.fetch(:land)
-        draw_country(gc, country, width, bg)
-        gc.fill(options.fetch(:land))
+    def template
+      template_filename = 'worldize/template.mvg.erb'
+      template_path = File.join(File.dirname(__FILE__), template_filename)
+      ERB.new(File.read(template_path), nil, '-')
+    end
+
+    # MiniMagick cannot ha
+    # rubocop:disable Metrics/MethodLength
+    def convert_mvg_to_png(input_file, width, ymin, ymax)
+      temp = Tempfile.new(['worldize', '.png'])
+      begin
+        # workaround, there isnt a way to enforce `mvg:filename` decode
+        MiniMagick::Tool::Convert.new do |convert|
+          convert << "mvg:#{input_file.path}"
+          convert << temp.path
+        end
+        png = MiniMagick::Image.open(temp.path, '.png')
+        png.combine_options do |c|
+          c.crop "#{width}x#{ymax - ymin}+0+#{ymin}!"
+        end
+      ensure
+        temp.close
+        temp.unlink # deletes the temp file
       end
```

Comment:
I believe in this code you are just using the name that `Tempfile` library have provided?.. Then probably it is better to just do
```ruby
temp_path = Dir::Tmpname.create(%w[worldize .png]) {} # yes, it has a bit weird call requirements with mandatory block
```